central/debug/service/service.go

```diff
@@ -651,6 +661,7 @@ func (s *serviceImpl) getDiagnosticDump(w http.ResponseWriter, r *http.Request)
 		withCPUProfile:    false,
 		withLogImbue:      true,
 		withAccessControl: true,
+		withCentral:       !env.DisableCentralDiagnostics.BooleanSetting(),
```

Comment:
No strong feelings either way, but IMHO, this is not a per-request property but rather a question of what mode the debug service runs in. And therefore I think this flag should be a field of `serviceImpl` and be passed in the `service.New(...)` function at its invocation in `main()`.

Reply:
I thought about it originally but I moved it to the debugDumpOptions since this service covers a lot more than just debug dump/diag bundle and this option doesn't apply to all. Just to these two. But also no strong feelings myself either, so I can switch it if you have a suggestion for a good property name?

Reply:
Naming is always hard, I would've suggested something like `mayAccessCentralEnv` or so, but maybe it's the simplest to just keep it.